app/models/druid_version_zip.rb

```diff
@@ -26,10 +27,63 @@ def file_path
     File.join(Settings.zip_storage, s3_key)
   end
 
-  # Currently computed live, soon will fetch from checksum storage
-  # @return [String] md5 base64-encoded checksum
-  # @todo Fetch (and cache) md5 from checksum storage
+  # WITHOUT (re)digesting the file, convert a hexdigest MD5 value to base64-endcoded equivalent.
+  # Motivation: we store hexdigest and S3 requires base64 in headers.
+  # @param [String] hex
+  # @return [String] base64 equivalent
+  def hex_to_base64(hex)
+    [[hex].pack("H*")].pack("m0")
+  end
+
+  # @return [Digest::MD5] cached md5 object
   def md5
-    @md5 ||= Digest::MD5.file(file_path).base64digest
+    @md5 ||= Digest::MD5.file(file_path)
+  end
+
+  # @return [Hash<Symbol => [String, Integer]>] metadata to accompany Zip file to an endpoint
+  def metadata
+    { checksum_md5: hexdigest, size: size, zip_cmd: zip_command, zip_version: zip_version }
+  end
+
+  def moab_version_path
+    @moab_version_path ||= Moab::StorageServices.object_version_path(druid.id, version)
+  end
+
+  # @return [Integer] Zip file size
+  def size
+    @size ||= FileTest.size(file_path)
+  end
+
+  # @return [String] shell command to unzip
```

Comment:
I leave that open.  The actual unzipping might execute in different places (e.g. bulk scenario vs. singular).

Reply:
i guess what I'm really asking is ... what is the purpose of including the commented out method here.

Reply:
After all, the zipping execution itself doesn't happen here, it happens in `ZipMaker`.

Reply:
I can remove the comment if you prefer, but since I mentioned being motivated by the idea of having the zip and unzip commands testable in the same place, I added it to reflect that.